helixbass/tsc-rust
# Emit type error

In this PR:
- emit a type error when parsing + type-checking the prefix unary expression `++true`

To test:
If you have a file named eg `tmp.tsx` in the repo root directory whose contents are just `++true` (no trailing newline) and run `cargo run tmp.tsx` you should see debug-logging of the `SourceFile` and then a single `DiagnosticWithLocation` associated with the `SourceFile` whose `file_name` is `"tmp.tsx"` and with location data `start: 2, length: 4` (ie the location of the `true` operand)
